gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/util/AttributeMetadataUtil.java

```diff
@@ -55,8 +57,34 @@ public static String getTimestampAttributeId(
   }
 
   private static String getStartTimeAttributeKeyName(String attributeScope) {
-    String timestamp = TimestampConfigs.getTimestampColumn(attributeScope);
-    return timestamp == null ? START_TIME_ATTRIBUTE_KEY : timestamp;
+    return Optional.ofNullable(TimestampConfigs.getTimestampColumn(attributeScope))
+        .orElse(START_TIME_ATTRIBUTE_KEY);
+  }
+
+
+  public static String getSpaceAttributeId(
+      AttributeMetadataProvider attributeMetadataProvider,
+      RequestContext requestContext,
+      String attributeScope) {
+    // Trace types have a space ID attribute
+    if (List.of(
```

Comment:
nit. Make this a set and probably extract out as static constant?

Reply:
Sure